aws/resource_aws_msk_cluster.go

```diff
@@ -35,6 +36,10 @@ func resourceAwsMskCluster() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"broker_client_vpc_ip_addresses": {
```

Comment:
i think it would be more convenient to have this as a list (or rather set)

Reply:
Thanks, looking into this. Would the suggested schema be a TypeSet which consists of N TypeString elements having the same attribute name?

If the following meets the intent, I'll look for an example on how to allow multiple attributes with the same name.
```
"broker_client_vpc_ip_addresses.<HASH>.ip.#": "3",
"broker_client_vpc_ip_addresses.<HASH>.ip.0": "10.0.0.1",
"broker_client_vpc_ip_addresses.<HASH>.ip.1": "10.0.0.2",
"broker_client_vpc_ip_addresses.<HASH>.ip.2": "10.0.0.3",
```

Reply:
I want to say yes but on another look, API is a little messed up. i wondering if it should not be under a node_info block and not stand alone attribute as there are other things like ENIs that can be also added as computed attributes.

Ideally it would be part of the response from the BrokerNodeGroupInfo in the read function and be under broker_node_group_info block but that's not how the API is structured.

what do you think?

Reply:
Agreed, the API makes the goal of obtaining all ip addresses for a cluster tricky. It feels like some changes to the MSK API would allow broker details to be retrieved more elegantly as you've described. Without API changes though, the current implementation seemed like the most practical approach to achieve the goal of obtaining the full list of broker ip addresses. One rationale for going with a comma separated string is to maintain parity with the similar outputs which already exist, bootstrap_brokers and bootstrap_brokers_tls. For obtaining additional cluster details which are part of BrokerNodeGroupInfo, a separate PR to provide a new object with a full schema may be appropriate. Please let me know if the compromise of a string for this particular use-case is reasonable with the current state of the API and existing outputs.

Reply:
I don't think that a comma separated makes sense here. It makes sense for broker addresses because thats usually how Kafka clients take them as inputs.
Vpc ip addresses are probably meant to be used in security groups and then you will need to split them again.

Whether it's a set of strings or set of Node info blocks with an IP attribute will let you pass it better.

Reply:
Makes sense, I'll look into using a set.

Reply:
Changed to use a set in https://github.com/terraform-providers/terraform-provider-aws/pull/12821/commits/400b144dafc05cacfc6acd23f76a4b684a00bbb9